csharp/extractor/Semmle.Extraction.CSharp/Entities/Method.cs

```diff
@@ -243,7 +243,12 @@ public static void AddExplicitInterfaceQualifierToId(Context cx, EscapingTextWri
             if (methodKind == MethodKind.ExplicitInterfaceImplementation)
             {
                 // Retrieve the original method kind
-                methodKind = methodDecl.ExplicitInterfaceImplementations.Select(m => m.MethodKind).FirstOrDefault();
+                if (methodDecl.ExplicitInterfaceImplementations.IsEmpty)
+                {
+                    throw new InternalError(methodDecl, $"Couldn't get the original method kind for an explicit interface implementation");
```

Comment:
nit: you can remove the `$`.

Reply:
Or, better, mention `methodDecl` in the error message.

Reply:
There's no need to mention it. It's added automatically: https://github.com/github/codeql/pull/13384#discussion_r1221164475